### ISSUE TITLE
Scorecard generation

### ISSUE THREAD

**Issue by jfly** [cubing/ccm#67]
James just informed me that people actually _do_ use this feature of cubecomps. I guess Berkeley comps don't because they never have printers.

Unfortunately, large competitions such as US Nationals can't use cubecomps to print out scorecards, because Tim has built this heat generation infrastructure.

Unfortunately, I think HTML/CSS isn't the best for generating printable pages. Lucas Garron has experience with this from Mark2 (https://github.com/cubing/mark2/issues/11), I'll ask him to comment here. If HTML/CSS is not an option, we can look into generating a PDF with javascript ([jspdf](https://parall.ax/products/jspdf)?)

If we support scorecards, then we could also assign competitors id numbers and print them on the scorecards. This would give an alternative way of choosing competitors when doing data entry.

**Comment by jbcm627** [comment]
This exists too:
https://github.com/pascoual/meteor-pdfkit

**Comment by jfly** [comment]
Neat idea from Pedro: scorecards could include the cutoffs for a round.

**Comment by lgarron** [comment]
> Neat idea from Pedro: scorecards could include the cutoffs for a round.

Sounds like a great idea, as long as we can express the different cutoffs properly.

**Comment by jfly** [comment]
Stealing an idea from James Malloy on the delegate list:

> In the UK we had a bit of confusion recently and cleared it up with everyone. All extra attempts were done correctly the past weekend in Edinburgh of which there were around 10 or so.

> Let's say 3rd attempt required extra, we would get the judge to write Extra in the box, both competitor and judge sign. And just take the scorecard back to the scramble table and just continue with the 4th, 5th etc. Then after the 5th solve they go and THEN do the extra solve, which is input in box 6. During data entry we would put the results in the order 1, 2, E, 4, 5.

> If you use computer scrambles and do the extra one straight away, you have to scroll down the list to get to the extra one, slowing down the scrambling and the competition.

> James

**Comment by lgarron** [comment]
Also sounds like a neat idea.

However, we should make sure that the attempt using scramble #4 shows up as fourth on the WCA (so that we can continue using the competition scrambles to investigate incidents with as little extra confusion as possible).

**Comment by jbcm627** [comment]
Yeah, this is a good idea. @lgarron, I think the procedure described handles your concern.  However I would prefer not to include entire extra lines for times if there is another way to achieve this.

While digging through emails, I also came across another idea: for competitions with multiple rooms, stages, or competition areas, put the room on the scorecard (related to #28).

Another idea I recall being implemented somewhere, but I can't seem to find: provide a graphical representation of penalties on the scorecard.

**Comment by jfly** [comment]
From the Back To Cubing Hong Kong 2015 delegate report:

![image](https://cloud.githubusercontent.com/assets/277474/6837655/1f74ebc4-d30c-11e4-89e9-bdf70af7c904.png)

**Comment by jbcm627** [comment]
Also from the report, the actual scorecards:
![scorecard](https://cloud.githubusercontent.com/assets/964048/6837681/b8cb506a-d325-11e4-852f-31308787212d.png)

**Comment by FatBoyXPC** [comment]
What the example cubecomps is missing is there is an extra number on the scorecard, in the top left. This is the nth person in that round of an event. This makes sorting easier for some people, such as @KitClement.

Example: http://imgur.com/RPbv9Ep

**Comment by jfly** [comment]
Hmm.. "nth person" means based on previous wca results? Like the psych sheet?

**Comment by FatBoyXPC** [comment]
For round 1, it at least means alphabetically. When you open a 2nd round of an event, and advance the competitors, when everybody has no results yet it still lists them alphabetically. I would assume in this case it still means alphabetically.

**Comment by Ranzha** [comment]
It was brought up here (https://github.com/cubing/icons/issues/15) that the Chinese penalty icons don't make clear whether penalties for having contact with the puzzle when operating the timer (A4b1, A6c) and hand placement while operating the timer (A4b, A6c) are being applied at the start or the end of an attempt.

To rectify this, as well as remain consistent with A7b1, I propose that scorecards have columns for starting penalties, the displayed result, ending penalties, and final result. This follows the "X + T + Y = F" format of A7b1 and keeps the (identical) icons for A4b1 and A6c separate.

**Comment by jfly** [comment]
Another cool scorecard layout (came up in the delegate report for Virginia Open Winter 2017 when discussing judges who assign extra scrambles unbeknownst to the delegate):

![image](https://cloud.githubusercontent.com/assets/277474/23189074/141ad0b6-f846-11e6-92d2-ff64edc18bc7.png)

**Comment by coder13** [comment]
Looks neat.
I'd probably remove "Category" and "Round" from up top and just have that row say "Pyraminx - Round 1" as it's fairly obvious to know the pyraminx is the category (event) and it says the round cleanly. 

With how uncommon extra scrambles are (although they do happen every competition), extra scramble reasons would be better to write on the back imo. I do like the design though.

**Comment by Ranzha** [comment]
An interesting idea would be to have customisable scorecards with optional
fields (i.e. check a box and get that bottom field).

Of course, implementation is a different story.

**Comment by lgarron** [comment]
> An interesting idea would be to have customisable scorecards with optional
> fields (i.e. check a box and get that bottom field).

Customizability has a cost.

I think we got a lot of benefit out of pretty much everyone using the same FMC sheet.

As long as it serves the purposes most competitions need, I think there's great value in TNoodle providing scorecards that are used the same way around the entire world.
